### PR TITLE
[Masternode] Migrate Powershell Script to .NET Console app

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Api/NodeController.cs [hunk 1]
```diff
@@ -71,6 +71,8 @@ public class NodeController : Controller
         /// <summary>An interface implementation used to retrieve unspent transactions.</summary>
         private readonly IGetUnspentTransaction getUnspentTransaction;
 
+        private readonly IInitialBlockDownloadState initialBlockDownloadState;
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.Api/NodeController.cs:74]: This got added for the IBD check when calling node status from the masternode registration app.

File: src/Stratis.Bitcoin.Features.PoA/Models/FederationMemberModel.cs [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-using NBitcoin;
 using Newtonsoft.Json;
 
 namespace Stratis.Bitcoin.Features.PoA.Models
 {
     public class FederationMemberModel
     {
         [JsonProperty("pubkey")]
-        public PubKey PubKey { get; set; }
+        public string PubKey { get; set; }
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Features.PoA/Models/FederationMemberModel.cs:9]: This doesnt need serialized as a `PubKey` instance.

File: src/Stratis.Features.FederatedPeg/FederatedPegFeature.cs [hunk 3]
```diff
@@ -284,7 +283,6 @@ public static IFullNodeBuilder AddFederatedPeg(this IFullNodeBuilder fullNodeBui
                         {
                             services.AddSingleton<IRewardDistributionManager, RewardDistributionManager>();
                             services.AddSingleton<ICoinbaseSplitter, PremineCoinbaseSplitter>();
-                            services.AddSingleton<BlockDefinition, FederatedPegBlockDefinition>();
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/FederatedPegFeature.cs:287]: This was a bug where when the `AddFederatedPeg` method add the `FederatedPegBlockDefinition` it gets overwritten with `AddPoACollateralMiningCapability` which adds the `SmartContractPoABlockDefinition`

File: src/Stratis.External.MasternodeRegistration/RegistrationService.cs [hunk 4]
```diff
@@ -0,0 +1,523 @@
...
+    {
+        private string rootDataFolder;
+        private Network mainchainNetwork;
+        private Network sidechainNetwork;
+        private const string nodeExecutable = "Stratis.CirrusMinerD.exe";
+
+        private const int CollateralRequirement = 100_000;
+        private const int FeeRequirement = 500;
+
+        private const string DataDir = @"D:\Stratis\StraxRegTest\Masternode";
```
Comment (zeptin) [inline src/Stratis.External.MasternodeRegistration/RegistrationService.cs:35]: This probably shouldn't be hardcoded?

Comment (fassadlr) [reply src/Stratis.External.MasternodeRegistration/RegistrationService.cs:35]: Fixed 👍

**Review by zeptin — APPROVED** [review]

**Pull request merged**